expat/expat_config.h.cmake

```diff
@@ -108,6 +108,19 @@
 #  define __func__ __FUNCTION__
 #endif
 
+/* Attribute to annotate explicit fallthrough. As noted on
+   https://gcc.gnu.org/onlinedocs/cpp/_005f_005fhas_005fattribute.html ,
+   combining the two tests in one conditional would not be portable. */
+#if defined __has_attribute
+#  if __has_attribute (fallthrough)
+#    define FALLTHROUGH __attribute__ ((fallthrough))
+#  else
+#    define FALLTHROUGH
+#  endif
+#else
+  #define FALLTHROUGH
+#endif
+
```

Comment:
Looks good in general; two things:
- We have two build systems — CMake and GNU Autotools; file `expat_config.h.cmake` only affects the CMake one and Autotools equivalent `expat_config.h.in` is generated (by autoheader, autoreconf, `./buildconf.sh`) and hence would not be a good option; plus rendered `expat_confg.h` is installed, hence this file is not the best place to put the new internal macro. From a quick look, probably file `lib/internal.h` is the best match — we already have sibling macro `UNUSED_P` in there.
- The [clang docs](https://clang.llvm.org/docs/LanguageExtensions.html#has-attribute) have an idea to avoid the duplication in a portable way. Their approach seems slightly less complex to me. What do you think about their approach?

Reply:
Thanks for the guidance on where to put the macro, I wasn't sure.  Moved.

I had considered and avoided the clang doc route because I felt leery of #defining a double-underscore symbol, but you're right that it's more readable.  Changed.